services/user-svc/internal/hwkc/client.go

```diff
@@ -0,0 +1,243 @@
+		http:            config.Client(ctx),
+	}
+
+	return client, err
+}
+
+func (c *Client) validateResponse(res *http.Response) error {
+	if res.StatusCode >= 200 && res.StatusCode <= 299 {
+		return nil
+	}
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return fmt.Errorf("response status code not 2xx, cannot read response body for error message: %w", err)
+	}
+	return fmt.Errorf("response status for %s - %s is %d: %s", res.Request.Method, res.Request.URL.String(), res.StatusCode, string(body))
+}
+
+func (c *Client) GetUserById(userID uuid.UUID) (*User, error) {
+	u := c.adminApiBaseUrl.JoinPath("users", userID.String())
+
+	res, err := c.http.Get(u.String())
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if err := c.validateResponse(res); err != nil {
+		return nil, err
+	}
+
+	var user *User
+	if err := json.NewDecoder(res.Body).Decode(&user); err != nil {
+		return nil, err
+	}
+
+	return user, nil
+}
+
+func (c *Client) GetOrganizationsForUserById(userID uuid.UUID) ([]*Organization, error) {
+	u := c.realmBaseUrl.JoinPath("users", userID.String(), "orgs")
+
+	res, err := c.http.Get(u.String())
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if err := c.validateResponse(res); err != nil {
+		return nil, err
+	}
+
+	var organizations []*Organization
+	if err := json.NewDecoder(res.Body).Decode(&organizations); err != nil {
+		return nil, err
+	}
+
+	for _, organization := range organizations {
+		organization.ParseRawAttributes()
+	}
+
+	return organizations, nil
+}
+
+func (c *Client) CreateOrganization(name, displayName string, isPersonal bool) (*Organization, error) {
+	u := c.realmBaseUrl.JoinPath("orgs")
+
+	payload := Organization{
+		Name:        hwutil.StrPtr(name),
+		DisplayName: hwutil.StrPtr(displayName),
+		RawAttributes: &map[string][]string{
+			"isPersonal": {strconv.FormatBool(isPersonal)},
+		},
+	}
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.http.Post(u.String(), "application/json", bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if err := c.validateResponse(res); err != nil {
+		return nil, fmt.Errorf("cannot create, invalid response: %w", err)
+	}
+
+	res, err = c.http.Get(res.Header.Get("Location"))
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if err := c.validateResponse(res); err != nil {
+		return nil, fmt.Errorf("cannot get created organization, invalid response: %w", err)
+	}
+
+	var organization Organization
+	if err := json.NewDecoder(res.Body).Decode(&organization); err != nil {
+		return nil, fmt.Errorf("cannot parse organization: %w", err)
+	}
+	organization.ParseRawAttributes()
+
+	return &organization, nil
+}
+
+func (c *Client) UpdateOrganization(organizationID uuid.UUID, organization Organization) error {
+	u := c.realmBaseUrl.JoinPath("orgs", organizationID.String())
+
+	jsonPayload, err := json.Marshal(organizationID)
```

Comment:
that's three lines too much for this

Reply:
Yes. But for now thats a simple Keycloak client. We can and probably need to expand this in the future.

Reply:
good thing I was being pedantic then c: